hudi-common/src/main/java/org/apache/hudi/common/table/view/RocksDbBasedFileSystemView.java

```diff
@@ -544,10 +574,13 @@ protected Option<HoodieInstant> getReplaceInstant(final HoodieFileGroupId fileGr
   }
 
   private Stream<HoodieFileGroup> getFileGroups(Stream<FileSlice> sliceStream) {
+    // HoodieFileGroup using timeline to determine whether a file slice is committed or not so passing
+    // completedAndCompactionTimeline object while creating HoodieFileGroup.
+    HoodieTimeline completedAndCompactionTimeline = getVisibleCommitsAndCompactionTimeline();
```

Comment:
may be we shd name this
completedWriteAndCompactionTimeline

Reply:
Refactored it.